src/module.c

```diff
@@ -9312,8 +9338,10 @@ void moduleFireServerEvent(uint64_t eid, int subid, void *data) {
                     selectDb(ctx.client, fi->dbnum);
             } else if (eid == REDISMODULE_EVENT_MODULE_CHANGE) {
                 RedisModule *m = data;
-                if (m == el->module)
+                if (m == el->module) {
+                    moduleFreeContext(&ctx);
```

Comment:
this is a very minor leak, but still for completion, i think we should mention it in the PR top comment.

Reply:
This is not a leak actually. It was using `moduleFreeContextReusedClient` previously. So, no need to call moduleFreeContext() previously.  

Now we allocate a temp client object from pool and return it to the pool. To do that, need to call moduleFreeContext().